src/lib.rs

```diff
@@ -103,9 +103,22 @@ pub enum DecodeFailed {
     /// An invalid block type was found.
     InvalidBlock(u8),
 
+    /// An invalid block size was found.
+    InvalidBlockSize(u32),
+
     /// An invalid pretree element was found.
     InvalidPretreeElement(u16),
 
+    /// Invalid pretree run-length encoding.
+    // TODO: Better name?
+    InvalidPretreeRLE,
```

Comment:
With Rust's naming convention, this would be:
```suggestion
    InvalidPretreeRle,
```

Reply:
Okay - let me fix that!